Python/pywarpx/diagnostics/data/BaseReader.py

```diff
@@ -0,0 +1,185 @@
+import collections
```

Comment:
As WarpX is setup now, perhaps a better place for these Python scripts would be in `Tools/PostProcessing`? Also, the more descriptive `reduced_data` would be a better name for the subdirectory instead of `data`.

This would be a separate PR, but I think it would be good if all of the post-processing scripts, including these, would be installed as part of the Python interface.

Reply:
Yes, the strategy is as follows:
These tools are part of the default Python install, and thus well placed here. But we will also allow to install them separately, for the app purists.